relay-server/src/actors/outcome.rs

```diff
@@ -693,30 +687,19 @@ impl KafkaOutcomesProducer {
     /// If the given Kafka configuration parameters are invalid, or an error happens during
     /// connecting during the broker, an error is returned.
     pub fn create(config: &Config) -> Result<Self, ServerError> {
-        let mut reused_producers = BTreeMap::new();
-        let producers = KafkaOutcomesProducer {
-            default: store::make_producer(config, &mut reused_producers, KafkaTopic::Outcomes)?,
-            billing: store::make_producer(
-                config,
-                &mut reused_producers,
-                KafkaTopic::OutcomesBilling,
-            )?,
-        };
-
-        Ok(producers)
-    }
-
-    /// Returns the producer for default outcomes.
-    pub fn default(&self) -> &Producer {
-        &self.default
-    }
-
-    /// Returns the producer for billing outcomes.
-    ///
-    /// Note that this may return the same producer instance as [`default`](Self::default) depending
-    /// on the configuration.
-    pub fn billing(&self) -> &Producer {
-        &self.billing
+        let mut client_builder = KafkaClient::builder();
+
+        for topic in &[KafkaTopic::Outcomes, KafkaTopic::OutcomesBilling] {
+            let kafka_config = &config
+                .kafka_config(*topic)
+                .map_err(|_| ServerErrorKind::KafkaError)?;
+            client_builder = client_builder
+                .add_kafka_topic_config(kafka_config)
+                .map_err(|_| ServerErrorKind::KafkaError)?;
```

Comment:
This nit still stands, but it's not a blocker: 

> I would give KafkaError a #[cause] ProducerError to propagate the information.

https://github.com/getsentry/relay/pull/1563/files/3050fca26247f6d2e61d77c2b74842a93a721465#r1013800430

Reply:
@jjbayer so, the reason that the error neither here nor in any other case when the `ServerErrorKind` is used is that, the enum contains all the `"empty"` variants and also implement `Copy` and `Clone` and also `Hash` where most of the error do not do that, see:
https://github.com/getsentry/relay/blob/1e2bbf96936adfd170dfdef2c7f4656eaaf30ea2/relay-server/src/service.rs#L41-L75

In case we want to propagate the errors - we would need  either to implement `Hash` trait on kafka `ClientError`  and remove `Clone` and `Copy`,  or if we add only additional formatted message inside the variant - we would have to at least remove `Copy` and `Clone`  and return reference in https://github.com/getsentry/relay/blob/1e2bbf96936adfd170dfdef2c7f4656eaaf30ea2/relay-server/src/service.rs#L96

What do you think would be better?

Reply:
I see. Ignore my comment then, given that this error only happens on initialization I think we can live without the extra information being bubbled up.